pkg/coscheduling/coscheduling.go

```diff
@@ -85,9 +95,43 @@ func (cs *Coscheduling) Name() string {
 // New initializes a new plugin and returns it.
 func New(_ *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
 	podLister := handle.SharedInformerFactory().Core().V1().Pods().Lister()
-	return &Coscheduling{frameworkHandle: handle,
+	cs := &Coscheduling{frameworkHandle: handle,
 		podLister: podLister,
-	}, nil
+		clock:     util.RealClock{},
+	}
+	podInformer := handle.SharedInformerFactory().Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch t := obj.(type) {
+				case *v1.Pod:
+					return !assignedPod(t)
+				case cache.DeletedFinalStateUnknown:
+					if pod, ok := t.Obj.(*v1.Pod); ok {
+						return !assignedPod(pod)
+					}
+					return false
+				default:
+					return false
+				}
+			},
+			Handler: cache.ResourceEventHandlerFuncs{
+				DeleteFunc: cs.markPodGroupAsExpired,
+			},
+		},
+	)
+	go wait.Until(cs.podGroupInfoGC, PodGroupGCInterval, nil)
+
+	return cs, nil
+}
+
+// FakeNew is used for test.
+func FakeNew(informerFactory informers.SharedInformerFactory, clock util.Clock) (*Coscheduling, error) {
+	cs := &Coscheduling{
+		clock: clock,
+	}
+	go wait.Until(cs.podGroupInfoGC, PodGroupGCInterval, nil)
```

Comment:
Do you want to replace nil with a stop channel, so that the UT won't leave the goroutine running forever.

Reply:
Done